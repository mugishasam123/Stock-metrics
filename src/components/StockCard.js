import PropTypes from 'prop-types';
import React from 'react';
import '../styles/card.css';
import { useNavigate } from 'react-router-dom';
import {
  IoIosArrowRoundForward,
  IoIosArrowRoundDown,
  IoIosArrowRoundUp,
} from 'react-icons/io';

const StockCard = ({ stock }) => {
  const navigate = useNavigate();
  return (
    <div
      className="stock-card"
      onClick={() => navigate(`/details/${stock.ticker}`)}
      aria-hidden="true"
    >
      <div className="company-name">
        <h3 className="title-name">{stock.companyName}</h3>
        <IoIosArrowRoundForward className="rounded" />
      </div>
      <span className="ticker">{stock.ticker}</span>
      <span>
        $
        {stock.price}
      </span>
      <span>
        <IoIosArrowRoundUp className="green-arrow" />
        {stock.changes}
      </span>
      <span>
        <IoIosArrowRoundDown className="red-arrow" />
        {stock.changesPercentage}
      </span>
    </div>
  );
};
StockCard.propTypes = {
  stock: PropTypes.string.isRequired,
  companyName: PropTypes.string.isRequired,
  ticker: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  changes: PropTypes.string.isRequired,
  changesPercentage: PropTypes.string.isRequired,
};

export default StockCard;
