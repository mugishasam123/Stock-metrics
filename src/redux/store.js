import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import { stocksListReducer, stockDetailsReducer } from './stocks/reducers/stockReducers';

const rootReducer = combineReducers({ stocks: stocksListReducer, details: stockDetailsReducer });
const store = createStore(rootReducer, applyMiddleware(thunk, logger));

export default store;
