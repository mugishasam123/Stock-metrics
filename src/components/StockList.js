import React, { useState } from 'react';

import { useSelector } from 'react-redux';
import StockCard from './StockCard';
import Search from './Search';
import '../styles/stocklist.css';
import searchFunc from '../utils/stockHelper';

const StockList = () => {
  const stocks = useSelector((state) => state.stocks);
  const [value, setValue] = useState('');

  const handleSearch = (e) => {
    setValue(e.target.value);
  };
  return (
    <>
      <section className="stocks">
        <Search handleSearch={handleSearch} data={value} />
        <div className="stock-list">
          {stocks
            .filter((stock) => searchFunc(stock, value))
            .map((stock) => (
              <StockCard key={stock.ticker} stock={stock} />
            ))}
        </div>
      </section>
    </>
  );
};

export default StockList;
