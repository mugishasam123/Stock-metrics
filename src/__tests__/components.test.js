import { render } from '@testing-library/react';
import { Router } from 'react-router';
import { Provider } from 'react-redux';
import { createMemoryHistory } from 'history';
import StockCard from '../components/StockCard';
import store from '../redux/store';
import StockList from '../components/StockList';

test('testing if stockcard is rendered', () => {
  const stock = {
    ticker: 'F',
    companyName: 'Ford',
    price: '32',
    changes: '0.23',
    changesPercentage: '0.003',
  };
  const history = createMemoryHistory();
  expect(
    render(
      <Router location={history.location} navigator={history}>
        <StockCard stock={stock} />
      </Router>,
    ),
  ).toMatchSnapshot();
});

test('if stockList renders correctly', () => {
  expect(
    render(
      <Provider store={store}>
        <StockList />
      </Provider>,
    ),
  ).toMatchSnapshot();
});
