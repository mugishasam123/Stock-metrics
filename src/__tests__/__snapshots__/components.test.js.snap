// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if stockList renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="stocks"
      >
        <div
          class="searchBar"
        >
          <input
            placeholder="Search..."
            type="text"
          />
          <svg
            class="search"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
            />
          </svg>
        </div>
        <div
          class="stock-list"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="stocks"
    >
      <div
        class="searchBar"
      >
        <input
          placeholder="Search..."
          type="text"
        />
        <svg
          class="search"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
          />
        </svg>
      </div>
      <div
        class="stock-list"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`testing if stockcard is rendered 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-hidden="true"
        class="stock-card"
      >
        <div
          class="company-name"
        >
          <h3
            class="title-name"
          >
            Ford
          </h3>
          <svg
            class="rounded"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M295.6 163.7c-5.1 5-5.1 13.3-.1 18.4l60.8 60.9H124.9c-7.1 0-12.9 5.8-12.9 13s5.8 13 12.9 13h231.3l-60.8 60.9c-5 5.1-4.9 13.3.1 18.4 5.1 5 13.2 5 18.3-.1l82.4-83c1.1-1.2 2-2.5 2.7-4.1.7-1.6 1-3.3 1-5 0-3.4-1.3-6.6-3.7-9.1l-82.4-83c-4.9-5.2-13.1-5.3-18.2-.3z"
            />
          </svg>
        </div>
        <span
          class="ticker"
        >
          F
        </span>
        <span>
          $
          32
        </span>
        <span>
          <svg
            class="green-arrow"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M348.3 216.4c-5 5.1-13.3 5.1-18.4.1L269 155.8v231.3c0 7.1-5.8 12.9-13 12.9s-13-5.8-13-12.9V155.8l-60.9 60.8c-5.1 5-13.3 4.9-18.4-.1-5-5.1-5-13.2.1-18.3l83-82.4c1.2-1.1 2.5-2 4.1-2.7 1.6-.7 3.3-1 5-1 3.4 0 6.6 1.3 9.1 3.7l83 82.4c5.2 4.9 5.3 13.1.3 18.2z"
            />
          </svg>
          0.23
        </span>
        <span>
          <svg
            class="red-arrow"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M348.3 295.6c-5-5.1-13.3-5.1-18.4-.1L269 356.2V124.9c0-7.1-5.8-12.9-13-12.9s-13 5.8-13 12.9v231.3l-60.9-60.8c-5.1-5-13.3-4.9-18.4.1-5 5.1-5 13.2.1 18.3l83 82.4c1.2 1.1 2.5 2 4.1 2.7 1.6.7 3.3 1 5 1 3.4 0 6.6-1.3 9.1-3.7l83-82.4c5.2-4.9 5.3-13.1.3-18.2z"
            />
          </svg>
          0.003
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-hidden="true"
      class="stock-card"
    >
      <div
        class="company-name"
      >
        <h3
          class="title-name"
        >
          Ford
        </h3>
        <svg
          class="rounded"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M295.6 163.7c-5.1 5-5.1 13.3-.1 18.4l60.8 60.9H124.9c-7.1 0-12.9 5.8-12.9 13s5.8 13 12.9 13h231.3l-60.8 60.9c-5 5.1-4.9 13.3.1 18.4 5.1 5 13.2 5 18.3-.1l82.4-83c1.1-1.2 2-2.5 2.7-4.1.7-1.6 1-3.3 1-5 0-3.4-1.3-6.6-3.7-9.1l-82.4-83c-4.9-5.2-13.1-5.3-18.2-.3z"
          />
        </svg>
      </div>
      <span
        class="ticker"
      >
        F
      </span>
      <span>
        $
        32
      </span>
      <span>
        <svg
          class="green-arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M348.3 216.4c-5 5.1-13.3 5.1-18.4.1L269 155.8v231.3c0 7.1-5.8 12.9-13 12.9s-13-5.8-13-12.9V155.8l-60.9 60.8c-5.1 5-13.3 4.9-18.4-.1-5-5.1-5-13.2.1-18.3l83-82.4c1.2-1.1 2.5-2 4.1-2.7 1.6-.7 3.3-1 5-1 3.4 0 6.6 1.3 9.1 3.7l83 82.4c5.2 4.9 5.3 13.1.3 18.2z"
          />
        </svg>
        0.23
      </span>
      <span>
        <svg
          class="red-arrow"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 512 512"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M348.3 295.6c-5-5.1-13.3-5.1-18.4-.1L269 356.2V124.9c0-7.1-5.8-12.9-13-12.9s-13 5.8-13 12.9v231.3l-60.9-60.8c-5.1-5-13.3-4.9-18.4.1-5 5.1-5 13.2.1 18.3l83 82.4c1.2 1.1 2.5 2 4.1 2.7 1.6.7 3.3 1 5 1 3.4 0 6.6-1.3 9.1-3.7l83-82.4c5.2-4.9 5.3-13.1.3-18.2z"
          />
        </svg>
        0.003
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
